### PR TITLE
Add Timestamp to json log as well

### PR DESCRIPTION
**Pull request by 7d4b9**
J'ai récupéré le format des timestamp qui était déjà utilisé dans le logger

### CODE REVIEW

**Comment by instabledesign** [comment]
Merci de votre contrib
Enfait les formatters de bases sont des formatters simple sans extra data ou extra fonctionnalités.
Par ailleurs vous pouvez atteindre le même but de 2 manières différentes
1️⃣ En developpant votre propre json formatter avant de l'injecter dans le `formatter.NewJSON(...)`
```go
import (
	"github.com/gol4ng/logger"
	"github.com/gol4ng/logger/formatter"
)

func JSONEncoderWithTimestamp(entry logger.Entry) ([]byte, error) {
	builder := &strings.Builder{}
	EntryToJSONWithTimestamp(entry, builder)

	return []byte(builder.String()), nil
}

func EntryToJSONWithTimestamp(entry logger.Entry, builder *strings.Builder) {
	builder.WriteRune('{')

	builder.WriteString("\"Message\":\"")
	builder.WriteString(entry.Message)
	builder.WriteString("\"")

	builder.WriteRune(',')
	builder.WriteString("\"Level\":")
	builder.WriteString(strconv.Itoa(int(entry.Level)))

	builder.WriteRune(',')
	builder.WriteString("\"Timestamp\":")
	builder.WriteString(strconv.FormatFloat(float64(time.Now().UnixNano())/1e9, 'f', 3, 64))

	builder.WriteRune(',')
	builder.WriteString("\"Context\":")

	formatter.ContextToJSON(entry.Context, builder)

	builder.WriteRune('}')
}

jsonLogger := logger.NewLogger(handler.Stream(os.StdOut, formatter.NewJSON(JSONEncoderWithTimestamp)))
```

2️⃣ En implementant votre propre middleware qui ajouteras la donnés direct sur le logger.Context
```go
func Timestampable() logger.MiddlewareInterface {
	return func(handler logger.HandlerInterface) logger.HandlerInterface {
		return func(entry logger.Entry) error {
			if entry.Context == nil {
				entry.Context = &logger.Context{}
			}
			entry.Context.
				Add("timestamp", strconv.FormatFloat(float64(time.Now().UnixNano())/1e9, 'f', 3, 64))
			return handler(entry)
		}
	}
}

jsonLogger := logger.NewLogger(handler.Stream(os.StdOut, formatter.NewJSON(JSONEncoderWithTimestamp)))
jsonLogger.Wrap(Timestampable())
```

Que pensez vous de ces alternatives?
Si vous souhaitez contribuer je me permet de vous renvoyer vers la [TODO list](https://github.com/gol4ng/logger#todo)

**Comment by rmasclef** [comment]
Please comment in English so everyone can understand the proposal and the answers 🙂

**Comment by rmasclef** [comment]
@instabledesign there's a little mistake on the second suggestion, you explain that you can either create a new formatter or a logger middleware

In the second snippet you use both the formatter and the middleware which is contradict with your comment ;)

Regarding the PR, I suggest we create a logger middleware that adds the timestamp to the logger context so it can be displayed in each formatter that displays the logger context

> TL;DR we could implement solution 2 in the lib 🙂

**Comment by 7d4b9** [comment]
In my mind both alternatives would work but as an absolute lazy coder and an absolute dummy in algorithm I would like to delegate to the logger library the way to make this in the optimal logger writer vision, as JSON by example, with timestamp or not, with call stack or not, etc...

Maybe it would be interesting to implement a `std` package with global logger and default config to be able to issue one line of log with one line of import and one line of `logger.Info(...sth...)`. I believe that it could be a nice feature and I may use it in production if it' ok.

I understand that the current version of the library does not provide this feature and that it may also not be inside the roadmap nor the vision of the creators.

I just would like to align the logger library in a [benchmark](https://github.com/7d4b9/golang-logger-bench) that give good results to the library and have been missing the `std` kind of logger.
My benchmark also would have wanted to reward friendly handling (call it UX) of the logger

**Comment by 7d4b9** [comment]
I understand and I absolutely agree with the choice not to put timestamp inside json by default:
* zap forces addition of the call stack
* zerolog, zap, logrus add timestamp by default also in JSON logs
* logger (this) does it in `GELF`encoder also
    * and I believed it was missing in JSON
       * I may be wrong so you can discard this PR if it's not in the vision 🙂

**Comment by instabledesign** [comment]
If you want you can implement solution :two: timestampable middleware with test suite, or i'll do it in few days.

> delegate to the logger library the way to make this in the optimal logger writer vision

I keep that in mind in order to add some helper to build appropriate logger

**Comment by rmasclef** [comment]
>     * logger (this) does it in `GELF`encoder also

It does indeed add a timestamp information for **GELF** formatter because that format actually requires it 👍 (see: https://docs.graylog.org/en/3.1/pages/gelf.html#gelf-payload-specification)

**Comment by 7d4b9** [comment]
I will use second solution. That way I will not have to define an extra layout and keep feeling of giving minimal information.

**Comment by rmasclef** [comment]
Don't hesitate to propose a pr in order to include the timestampMiddleware
